src/services/CertificateGenerationService.ts

```diff
@@ -807,6 +1039,183 @@ class CertificateGenerationService {
     return defectString;
   }
 
+  /**
+   * Returns a formatted welsh string containing data about a given defect
+   * @param defect - the defect for which to generate the formatted welsh string
+   * @param vehicleType - the vehicle type from the test result
+   * @param flattenedDefects - the list of flattened defects
+   */
+  public formatDefectWelsh(
+    defect: any,
+    vehicleType: any,
+    flattenedDefects: IFlatDefect[]
+  ) {
+    const toUpperFirstLetter: any = (word: string) =>
+      word.charAt(0).toUpperCase() + word.slice(1);
+
+    const filteredFlatDefects: IFlatDefect[] = flattenedDefects.filter(
+      (x: IFlatDefect) => defect.deficiencyRef === x.ref
+    );
+
+    const filteredFlatDefect: IFlatDefect | null = this.filterFlatDefects(
+      filteredFlatDefects,
+      vehicleType
+    );
+
```

Comment:
I think you'd probably want to kick it out at this point if no defect is found (the method below)          console.error("ERROR: Unable to find a filtered defect");
          return null;

This will also remove the need for the not null check on the below

Reply:
Not sure I follow? We wouldn't want to throw an error as that could impact the English cert getting generated in the event something goes wrong with the Welsh cert being generated. 

This is really just belts and braces, theoretically the defect and it's translation will always exist, as the defects get pulled down from the taxonomy, the same taxonomy that has the translations, and that is what is selectable by the user.